src/wallet/wallet.cpp

```diff
@@ -1416,7 +1423,7 @@ void CWallet::BlockConnected(const CBlock& block, const std::vector<CTransaction
     m_last_block_processed_height = height;
     m_last_block_processed = block_hash;
     for (size_t i = 0; i < block.vtx.size(); i++) {
-        SyncTransaction(block.vtx[i], CWalletTx::Status::CONFIRMED, block_hash, i);
+        SyncTransaction(block.vtx[i], CWalletTx::Status::CONFIRMED, block_hash, m_last_block_processed_height, i);
```

Comment:
In commit "Add block_height field in struct Confirmation" (de8fe7d717042823f2aa429be2a260904a10b42e)

Can you replace m_last_block_processed_height with height on this line? It's slightly worrying to depend on the wallet member if it might get updated for some reason, and also better to be more consistent with previous `block_hash` argument.